Added Splunk Eval cheatsheet

f33dy0urhe4d opened the pull request:
###### What does your Pull Request do (check all that apply)?

Choose the most relevant items and use the following title template to name
your Pull Request.

- [x] New Instant Answer
    - [x] Cheat Sheets: **`New Splunk Eval Functions Cheat Sheet`**

###### Description of changes

SPLEF lists all the Splunk Eval Functions in a matter of seconds, without requiring to browse to the specific Docs page every time..

---

Instant Answer Page: https://duck.co/ia/view/splunkevalfunctions_cheat_sheet

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @f33dy0urhe4d

daxtheduck commented:
@f33dy0urhe4d Hey!
        
Thanks for taking the time to contribute! We really appreciate it.

We work closely with every contributor to make Instant Answers the best they can be, so we appreciate your patience as we look over your code. From here, the process usually goes like this:

1. Pull Request is reviewed by the DuckDuckGo staff and community
2. Staff and community will leave feedback with any necessary updates to the function or design.
3. Once you've made any necessary corrections, then your Instant Answer will be merged and deployed live on DuckDuckGo!

If you have any questions along the way, feel free to ask them here. Our staff and community are also available on Slack to answer any questions you may have. If you'd like to join us there please head to https://quackslack.herokuapp.com/ to get an invite.

>More Info: http://docs.duckduckhack.com/submitting/submitting-overview.html

Thanks!

daxtheduck commented:
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

mintsoft commented:
@duckduckgo/duckduckhack-contributors Can anyone review this for content?

jophab commented:
@f33dy0urhe4d There are two files in this pr now.
Lets delete one of them and create another pr for that one

zachthompson commented:
@f33dy0urhe4d The file names need to match the ID inside the JSON (which should also match the ID in the instant answer page.)

f33dy0urhe4d commented:
Hello, thanks everyone for the review.
@jophab I've deleted one the useless file and committed once again
 @zachthompson I've changed the ID in the instant answer page

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Splunk eval cheatsheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/splunkevalfunctions_cheat_sheet_index.png?dhtXXEtE7z)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/splunkevalfunctions_cheat_sheet_mobile.png?dhtXXEtE7z)

jophab commented on share/goodie/cheat_sheets/json/splunkevalfunctions.json:
```diff
@@ -0,0 +1,366 @@
+{
+    "id": "splunkevalfunctions_cheat_sheet",
+    "name": "SplunkEvalFunctions",
```
Lets give a space between the words

jophab commented on share/goodie/cheat_sheets/json/splunkevalfunctions.json:
```diff
@@ -0,0 +1,366 @@
+{
+    "id": "splunkevalfunctions_cheat_sheet",
+    "name": "SplunkEvalFunctions",
+    "description": "Splunk evaluation functions cheatsheet",
```
Shall we have a description that describes the contents of the cheat sheet.?
Or else remove it, because it is redundant

jophab commented on share/goodie/cheat_sheets/json/splunkevalfunctions.json:
```diff
@@ -0,0 +1,366 @@
+    "name": "SplunkEvalFunctions",
+    "description": "Splunk evaluation functions cheatsheet",
+    "metadata": {
+        "sourceName": "Splunk Evaluation Functions",
```
"Splunk" would be enough here

jophab commented on share/goodie/cheat_sheets/json/splunkevalfunctions.json:
```diff
@@ -0,0 +1,366 @@
+        "Multivalue functions",
+        "Statistical functions",
+        "Text functions",
+        "Trigonometry and Hyperbolic functions"
```
Lets capitalize f in funtions at all section names

jophab commented on share/goodie/cheat_sheets/json/splunkevalfunctions.json:
```diff
@@ -0,0 +1,366 @@
+            },
+            {
+                "key": "if(X,Y,Z)",
+                "val": "... | eval err=if(error == 200, \"OK\", \"Error\")"
```
Instead of giving examples at val field, I would prefer a short description about the key field

f33dy0urhe4d replied:
This cheat sheet is to be intended with the true meaning of it. That is, I think that having a real-working, concise example at hand is much better than having a description for each command..

daxtheduck commented:
Automated screenshot from [beta.duckduckgo.com](https://beta.duckduckgo.com/?q=Splunk eval cheatsheet)
                ![](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/splunkevalfunctions_cheat_sheet_index.png?wda6Y0ADNC)
                [Mobile screenshot](https://images.duckduckgo.com/iu/?u=https://ia-screenshots.s3.amazonaws.com/splunkevalfunctions_cheat_sheet_mobile.png?wda6Y0ADNC)

GuiltyDolphin commented:
@jophab @f33dy0urhe4d How's this going?

jophab commented:
@GuiltyDolphin I think it is good to merge.
What do you say about the val field content?
Whether it should be a description or an example?

mintsoft commented:
@jophab I think this is essentially like a lookup table for commonly used blocks of code; so it's fine imo

daxtheduck commented:
@f33dy0urhe4d this is now merged! It should be live on DuckDuckGo within one week (or during the next release) and we'll be sure to notify you when it happens.

After your Instant Answer goes live for the World to use, you will be able to see [traffic information](http://docs.duckduckhack.com/maintaining/tools.html#monitor-your-instant-answer-page) about your IA as well as receive feedback directly from users. 
                
Please refer to the [maintenance guide and docs](http://docs.duckduckhack.com/maintaining/guidelines.html) to keep up with potential changes to your Instant Answer.
                
Thank you again for your help!

mintsoft commented:
Thanks @f33dy0urhe4d @jophab If we get feedback around the value then we can revisit it :+1:
